Home_Work13/HW13.cpp

```diff
@@ -0,0 +1,442 @@
+					if (mainModeChoise == "2")
+					{
+						wordOfTheDayGuessStatus = "true";
+						writeSaveData(today, guessWord, wordOfTheDayGuessStatus, saveDataToFile);
+					}
+					break;
+				}
+			}
+		}
+
+	}
+}
+
+void printMainMenu()
+{
+	std::cout << std::endl;
+	std::cout << "!Hello this is Wordl Game!" << std::endl;
+	std::cout << "For start game in usual mode Enter 1" << std::endl;
+	std::cout << "For start game in Word of the day mode Enter 2" << std::endl;
+	std::cout << "For exit the game Enter 0" << std::endl;
+	std::cout << "Your input: ";
+}
+
+void printGameStartedMenu(int attempts)
+{
+	std::cout << std::endl;
+	std::cout << "Yor word generated try to guess it in " << attempts << " attempts" << std::endl;
+	std::cout << "To start game enter your guess" << std::endl;
+	std::cout << "To return to presious menu enter 0" << std::endl;
+	std::cout << "If the letter in result is uppercase this mean it correctlly guessed" << std::endl;
+	std::cout << "If the letter in result is lowercase this mean it in wrong place" << std::endl;
+}
+
+std::string mainModeErrors(std::string mainModeChoise)
+{
+	std::string inputErrors = "";
+	
+	bool inputContainsLettersOrSpecialChar = std::any_of(mainModeChoise.begin(), mainModeChoise.end(), [](char a) {if (!std::isdigit(a)) { return true; } return false; });
+	bool lengthNotEqualOne = (mainModeChoise.length() != 1);
+	bool wrongNumber = (mainModeChoise != "0" && mainModeChoise != "1" && mainModeChoise != "2");
+
+	if (lengthNotEqualOne)
+	{
+		inputErrors = " Input has wrong length ";
+	}
+
+	else if (inputContainsLettersOrSpecialChar)
+	{
+		inputErrors = " Input contain letters or special characters ";
+	}
+
+	else if (wrongNumber)
+	{
+		inputErrors = " Wrong control value ";
+	}
+
+	return inputErrors;
+}
+
+std::string gameModeErrors(std::string inputString, int wordLength)
+{
+	std::string inputErrors = "";
+	bool inputContainsDigitsOrSpecialChar = std::any_of(inputString.begin(), inputString.end(), [](char a) {if (!std::isalpha(a)) { return true; } return false; });
+	bool lengthNotEqualFive = (inputString.length() != wordLength);
+	bool lengthNotEqualOne = (inputString.length() != 1);
+	bool wrongNumber = (inputString != "0");
+
+	if (lengthNotEqualFive && lengthNotEqualOne)
+	{
+		inputErrors = " Input has wrong length ";
+	}
+
+	else if (inputContainsDigitsOrSpecialChar && lengthNotEqualOne)
+	{
+		inputErrors = " Input contain digits or special characters ";
+	}
+
+	else if (wrongNumber && !lengthNotEqualOne && !inputContainsDigitsOrSpecialChar)
+	{
+		inputErrors = " Wrong control value ";
+	}
+
+	else if (wrongNumber && !lengthNotEqualOne && inputContainsDigitsOrSpecialChar)
+	{
+		inputErrors = "  Input contain digits or special characters ";
+	}
+
+	return inputErrors;
+}
+
+std::string generateWord(const std::string WordsFile, const int WordsCount, const int wordLengt)
+{
+	std::srand(std::time(nullptr)); //generate random number
```

Comment:
should be called only once. Calling it every generateWord( ) call may break randomness feel